control-plane/subcommand/gateway-resources/command.go

```diff
@@ -604,6 +636,35 @@ func exponentialBackoffWithMaxIntervalAndTime() *backoff.ExponentialBackOff {
 	return backoff
 }
 
+func getScrapePort(v string) (int, bool) {
+	port, err := strconv.Atoi(v)
+	if err != nil {
+		// we only use the port if it's actually valid
+		return 0, false
+	}
+	if port < 1024 || port > 65535 {
+		return 0, false
+	}
+	return port, true
+}
+
+func getScrapePath(v string) (string, bool) {
+	if v == "" {
+		return "", false
+	}
+	return v, true
+}
+
+func getMetricsEnabled(v string) (bool, bool) {
+	if v == "true" {
+		return true, true
+	}
+	if v == "false" {
+		return false, true
+	}
+	return false, false
+}
+
```

Comment:
It feels odd to me that we're validating in 2 places: once here for the managed gateway class and once in `control-plane/api-gateway/common/metrics.go` that applies to all consumed `GatewayClasses`.

I don't have super strong feelings about it but am more generally concerned about having two pieces of similar code to maintain if we ever need to change things. Just for your consideration.

Reply:
So, I suppose we can refactor this into some generic validation block that could be re-used between the different subcommands, but it is validating two different methods of input currently:

1. The raw helm values themselves.
2. Values that may come from helm, from the spec of the `GatewayClassConfig`, or from an annotation.

Maybe I can do a quick follow-up pass after merge to consolidate the validation? Just thinking that we could keep as-is for now if we need this merged by later today.

Reply:
I'm okay with keeping as-is for now, just wanna make sure we're intentional about it